samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java

```diff
@@ -113,13 +113,13 @@ public SamzaSqlApplicationConfig(Config staticConfig) {
     inputSystemStreamConfigBySource = queryInfo.stream()
         .map(QueryInfo::getSources)
         .flatMap(Collection::stream)
-        .collect(Collectors.toMap(Function.identity(), src -> ioResolver.fetchSourceInfo(src)));
+        .collect(Collectors.toMap(Function.identity(), ioResolver::fetchSourceInfo, (src1, src2) -> src1));
```

Comment:
How about we use `distinct()` before `collect()` to save some resolver calls to fetch source information?

Reply:
right. thanks for pointing this out.